src/components/App.tsx

```diff
@@ -1,15 +1,44 @@
+import { useState } from 'react'
+import { Modal } from './modal'
+
+const Body = ({setShowModal} : any) => {
+  return (
+    <>
+      <p>
+        In order to close the modal, you can click outside of it. Also, you can click the button
+        on the top right corner or click this button below
+      </p>
+      <button onClick={() => setShowModal(false)}>Click me!</button>
+    </>
+  )
+}
 
 const App = () => {
+  const [showModal, setShowModal] = useState(false)
+
+  const handleShowModal = () => {
+    setShowModal(true)
+  }
+
   return (
     <>
       <h1>Trabe interview</h1>
       <h2>Exercise 2: Modal</h2>
 
       <hr />
 
-    </>
+      <button onClick={handleShowModal}>Show modal</button>
 
-  );
+      <Modal 
+        showModal={showModal} 
+        setShowModal={setShowModal}
```

Comment:
The modal should not receive this, prop. Maybe a onClose handler, but definitely not this state setter :)

Reply:
Definitely! Now my props are these! 
<img width="314" alt="Captura de pantalla 2022-03-21 a las 13 08 35" src="https://user-images.githubusercontent.com/51116828/159249322-376ef21b-0e12-4601-b414-c0d92e224068.png">